### PR TITLE
RtlCopyMemory is not available on Nanoserver use builtin copy instead

### PR DESCRIPTION
**Pull request by darrenstahlmsft**
Signed-off-by: Darren Stahl <darst@microsoft.com>

### CODE REVIEW

File: pipe.go [hunk 1]
```diff
@@ -18,12 +18,10 @@ import (
 //sys waitNamedPipe(name string, timeout uint32) (err error) = WaitNamedPipeW
 //sys getNamedPipeInfo(pipe syscall.Handle, flags *uint32, outSize *uint32, inSize *uint32, maxInstances *uint32) (err error) = GetNamedPipeInfo
 //sys getNamedPipeHandleState(pipe syscall.Handle, state *uint32, curInstances *uint32, maxCollectionCount *uint32, collectDataTimeout *uint32, userName *uint16, maxUserNameSize uint32) (err error) = GetNamedPipeHandleStateW
-//sys localAlloc(uFlags uint32, length uint32) (ptr uintptr) = LocalAlloc
-//sys copyMemory(dst uintptr, src uintptr, length uint32) = RtlCopyMemory
 
 type securityAttributes struct {
```
Comment (jstarks) [inline pipe.go]: I wonder why we aren't just using syscall.SecurityAttributes.

File: pipe.go [hunk 2]
```diff
@@ -18,12 +18,10 @@ import (
...
 	Length             uint32
-	SecurityDescriptor uintptr
+	SecurityDescriptor []byte
```
Comment (jstarks) [inline pipe.go]: This won't work.

File: pipe.go [hunk 3]
```diff
@@ -237,9 +235,8 @@ func makeServerPipeHandle(path string, securityDescriptor []byte, c *PipeConfig,
 	sa.Length = uint32(unsafe.Sizeof(*sa))
 	if securityDescriptor != nil {
 		len := uint32(len(securityDescriptor))
-		sa.SecurityDescriptor = localAlloc(0, len)
```
Comment (jstarks) [inline pipe.go]: Still need the localAlloc, I think.

File: pipe.go (hunk 1 above)
Comment (darrenstahlmsft) [inline pipe.go]: Didn't know that existed. Using it now.

File: pipe.go (hunk 2 above)
Comment (darrenstahlmsft) [inline pipe.go]: It works. Likely due to a fluke, but this is removed now

File: pipe.go (hunk 3 above)
Comment (darrenstahlmsft) [inline pipe.go]: Wasn't thinking. Added it back.

**Comment by darrenstahlmsft** [comment]
This is closer to the correct fix here. I'm still unsure about the casts on `sa.SecurityDescriptor`, and `go vet` is complaining, but it currently works.

File: pipe.go [hunk 4]
```diff
@@ -233,13 +226,13 @@ func makeServerPipeHandle(path string, securityDescriptor []byte, c *PipeConfig,
...
 
-	sa := &securityAttributes{}
+	sa := &syscall.SecurityAttributes{}
 	sa.Length = uint32(unsafe.Sizeof(*sa))
 	if securityDescriptor != nil {
 		len := uint32(len(securityDescriptor))
 		sa.SecurityDescriptor = localAlloc(0, len)
 		defer localFree(sa.SecurityDescriptor)
-		copyMemory(sa.SecurityDescriptor, uintptr(unsafe.Pointer(&securityDescriptor[0])), len)
+		copy((*[0xffff]byte)(unsafe.Pointer(sa.SecurityDescriptor))[:len], (*[0xffff]byte)(unsafe.Pointer(&securityDescriptor[0]))[:len])
```
Comment (jstarks) [inline pipe.go]: You only need to cast the destination. The source is already a `[]byte` so it can be passed directly to `copy`.

Comment (darrenstahlmsft) [inline pipe.go]: Fixed

**Comment by jstarks** [comment]
LGTM

**Pull request merged**